### PR TITLE
DOC: add Returns and Examples for scipy.ndimage.correlate() docstring

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix a part of #7168

#### What does this implement/fix?
add Returns and Examples for scipy.ndimage.correlate() docstring

### CODE REVIEW

**Comment by rgommers** [comment]
Thanks @AtsushiSakai. Still failing, `unexpected indentation` on TravisCI.

**Comment by AtsushiSakai** [comment]
@rgommers I’m sorry. From the next time, i will use draft when my PR is not ready for review. I fix it. PATL.

**Comment by rgommers** [comment]
> @rgommers I’m sorry. From the next time, i will use draft when my PR is not ready for review.

No worries. If you expected it to pass CI it was perfectly fine as it was (I do the same), if you know it's not ready you can indeed use draft mode or title it WIP: ....

File: scipy/ndimage/filters.py [hunk 1]
```diff
@@ -643,9 +643,45 @@ def correlate(input, weights, output=None, mode='reflect', cval=0.0,
...
+        The result of correlation of `input` with `weights`.
+
     See Also
     --------
     convolve : Convolve an image with a kernel.
+
+    Examples
+    --------
+    Correlation is the process of moving a filter mask often referred to
+    as kernel over the image and computing the sum of products at each location.
```
Comment (rgommers) [inline scipy/ndimage/filters.py:658]: This needs a blank line here to render correctly.

File: scipy/ndimage/filters.py [hunk 2]
```diff
@@ -643,9 +643,45 @@ def correlate(input, weights, output=None, mode='reflect', cval=0.0,
...
+
+    Define a kernel (weights) for correlation. In this example, it is for sum of
+    center and up, down, left and right next elements.
+
+    >>> weights = [[0, 1, 0],
+    ...            [1, 1, 1],
+    ...            [0, 1, 0]]
+
+    We can calculate a correlation result:
+    For example, element [2,2] is 7 + 11 + 12 + 13 + 17 = 60
```
Comment (rgommers) [inline scipy/ndimage/filters.py:676]: Could you use double backticks around `[2,2]` and the `7 + ...` expression? Will look nicer in the rendered html.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @AtsushiSakai, nice example. Just a couple of formatting issues.

**Comment by rgommers** [comment]
Okay this was definitely fixed, so better not to switch to draft mode. I want to push the green button but can't right now.

**Comment by AtsushiSakai** [comment]
@rgommers Sorry, go ahead.

**Comment by ilayn** [comment]
Ralf you can get it out of draft mode by clicking the button "Ready for review"

**Comment by rgommers** [comment]
> Ralf you can get it out of draft mode by clicking the button "Ready for review"

ah thanks, never done that before

**Comment by rgommers** [comment]
Merged, thanks @AtsushiSakai

**Pull request merged**